### PR TITLE
AC-1072: Add APIs for Encounter functions to the SDK

### PR DESCRIPTION
**Pull request by shubhamsgit**
## Description of what I changed

- New APIs for Encounters like fetching and saving Encounter Resources
- Created functions for Encounter DAO for saving and deleting records
- Changed schema of entities
- Created new entities

## Issue I worked on

JIRA Issue: https://issues.openmrs.org/browse/AC-1072

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

- [ ] I have **added tests** to cover my changes. (If you refactored
existing code that was well tested you do not have to add tests)

- [x] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

File: openmrs-android-sdk/src/main/java/com/openmrs/android_sdk/library/databases/entities/StandaloneEncounterEntity.kt [hunk 1]
```diff
@@ -0,0 +1,78 @@
+package com.openmrs.android_sdk.library.databases.entities
```
Comment (rishabh-997) [inline openmrs-android-sdk/src/main/java/com/openmrs/android_sdk/library/databases/entities/StandaloneEncounterEntity.kt:1]: Add copywrights in all the new files please

File: openmrs-android-sdk/src/main/java/com/openmrs/android_sdk/library/databases/entities/StandaloneEncounterEntity.kt [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+ */
+@Entity(tableName = "standaloneEncounters")
+class StandaloneEncounterEntity : Resource {
+
+    @ColumnInfo(name = "visit_uuid")
+    var visitUuid: String? = null
+
+    @ColumnInfo(name = "encounterDatetime")
+    var encounterDateTime: String? = null
+    /**
```
Comment (rishabh-997) [inline openmrs-android-sdk/src/main/java/com/openmrs/android_sdk/library/databases/entities/StandaloneEncounterEntity.kt:19]: new line btw comments and var

File: openmrs-android-sdk/src/main/java/com/openmrs/android_sdk/library/databases/entities/StandaloneEncounterEntity.kt [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+     * Gets and sets encounter type.
+     *
+     * @return the encounter type
+     */
+    @ColumnInfo(name = "type")
+
```
Comment (rishabh-997) [inline openmrs-android-sdk/src/main/java/com/openmrs/android_sdk/library/databases/entities/StandaloneEncounterEntity.kt:25]: remove this line... this file needs reformattings, pls take care of that

File: openmrs-android-sdk/src/main/java/com/openmrs/android_sdk/library/databases/AppDatabaseHelper.kt [hunk 4]
```diff
@@ -20,13 +20,7 @@ import com.openmrs.android_sdk.library.api.repository.FormRepository
...
 import com.openmrs.android_sdk.library.dao.ObservationDAO
 import com.openmrs.android_sdk.library.dao.PatientDAO
-import com.openmrs.android_sdk.library.databases.entities.AllergyEntity
-import com.openmrs.android_sdk.library.databases.entities.ConceptEntity
-import com.openmrs.android_sdk.library.databases.entities.EncounterEntity
-import com.openmrs.android_sdk.library.databases.entities.LocationEntity
-import com.openmrs.android_sdk.library.databases.entities.ObservationEntity
-import com.openmrs.android_sdk.library.databases.entities.PatientEntity
-import com.openmrs.android_sdk.library.databases.entities.VisitEntity
+import com.openmrs.android_sdk.library.databases.entities.*
```
Comment (rishabh-997) [inline openmrs-android-sdk/src/main/java/com/openmrs/android_sdk/library/databases/AppDatabaseHelper.kt:23]: Pls avoid start imports

**Review by rishabh-997 — COMMENTED** [review]
Good work adding encounter API functions to SDK

File: openmrs-android-sdk/src/main/java/com/openmrs/android_sdk/library/databases/entities/StandaloneEncounterEntity.kt (hunk 2 above)
Comment (shubhamsgit) [reply openmrs-android-sdk/src/main/java/com/openmrs/android_sdk/library/databases/entities/StandaloneEncounterEntity.kt]: Looked strange to me too but I took a reference from other entities in the SDK created by Saurabh. They are important for the SDK documentation.

File: openmrs-android-sdk/src/main/java/com/openmrs/android_sdk/library/databases/entities/StandaloneEncounterEntity.kt (hunk 1 above)
Comment (shubhamsgit) [reply openmrs-android-sdk/src/main/java/com/openmrs/android_sdk/library/databases/entities/StandaloneEncounterEntity.kt:15]: Done

**Pull request merged**